test/units/modules/network/fortimanager/test_fmgr_device_group.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2018 Fortinet, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <https://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import json
+from pyFMG.fortimgr import FortiManager
+import pytest
+
+try:
+    from ansible.modules.network.fortimanager import fmgr_device_group
+except ImportError:
+    pytest.skip(
+        "Could not load required modules for testing",
+        allow_module_level=True)
+
+fmg_instance = FortiManager("1.1.1.1", "admin", "")
+
+
+def load_fixtures():
+    fixture_path = os.path.join(os.path.dirname(__file__), 'fixtures') + "/{filename}.json".format(
+        filename=os.path.splitext(os.path.basename(__file__))[0])
+    try:
+        with open(fixture_path, "r") as fixture_file:
+            fixture_data = json.load(fixture_file)
+    except IOError:
+        return []
+    return [fixture_data]
+
+
+@pytest.fixture(scope="function", params=load_fixtures())
+def fixture_data(request):
+    func_name = request.function.__name__.replace("test_", "")
+    return request.param.get(func_name, None)
+
+
+def test_add_device_group(fixture_data, mocker):
```

Comment:
@ftntcorecse in `add_device_group` module example:
```
- name: ADD DEVICES TO DEVICE GROUP
  fmgr_device_group:
    host: "{{inventory_hostname}}"
    username: "{{ username }}"
    password: "{{ password }}"
    state: "present"
    grp_name: "TestGroup"
    grp_members: "FGT1,FGT2"
    adom: "ansible"
    vdom: "root"
```
`Mode` arguments is not present also other args differ slightly. Is this as per expectation if yes please take care in future PRs that module example and tests least should be same.

Reply:
This per expectation. This was one of the very first modules that was written this year. We ditched "state" for "mode" soon after this, because it's a little less vague. If you like, I can re-write this and get rid of "state" all together. Just let me know, but otherwise, know that this is no longer our standard and this should be a one off.

Reply:
@ftntcorecse Thanks for the confirmation and yes it would be great if you can re-write the same and remove `state` altogether that ways complete fortinet modules shall be aligned with `mode` instead of `state`.

Reply:
Ok -- I should have this done by tomorrow EOB.

Reply:
Sorry for the delay on this -- Have removed state parameter in favor of mode and retested everything. This module is ready to ship.